micromanager_gui/explore_sample.py

```diff
@@ -132,6 +133,31 @@ def _on_explorer_frame(self, image: np.ndarray, event: useq.MDAEvent):
         self.viewer.reset_view()
 
     def _on_mda_finished(self, sequence: useq.MDASequence):
+        meta = self.SEQUENCE_META.get(sequence) or SequenceMeta()
+        seq_uid = sequence.uid
+
+        if meta.mode == "explorer":
+            ch_and_id = []
+            for layer in self.viewer.layers:
```

Comment:
i'm quite confused why we're iterating through `viewer.layers` twice here.  once to build a list of strings, and a second time to match layers the the strings.  Could this not be done in a single pass?

```py
layers = []
for layer in self.viewer.layers:
    if some_string in layer.name ...:
        layers.append(layer)
link_layers(layer_list)
```